kotlinx-coroutines-core/common/src/internal/Concurrent.common.kt

```diff
@@ -19,3 +19,22 @@ internal expect fun <E> identitySet(expectedSize: Int): MutableSet<E>
 @OptionalExpectation
 @Target(AnnotationTarget.FIELD)
 internal expect annotation class BenignDataRace()
+
+// Used **only** as a workaround for #3820 in StateFlow. Do not use anywhere else
```

Comment:
`@OptIn(StopDoNotUse::class)`? `class BadAtomicReferenceForWorkarounds`? An arbitrary version of me down the line can't be trusted not to use this `AtomicReference` accidentally, and I don't read non-KDoc comments above every entity I use.